snc.sh

```diff
@@ -239,11 +239,15 @@ retry ${OC} delete mc chronyd-mask
 # Wait for the cluster again to become stable because of all the patches/changes
 wait_till_cluster_stable
 
+mc_before_removing_pullsecret=$(retry ${OC} get mc --sort-by=.metadata.creationTimestamp --no-headers -oname)
 # Replace pull secret with a null json string '{}'
 retry ${OC} replace -f pull-secret.yaml
+mc_after_removing_pullsecret=$(retry ${OC} get mc --sort-by=.metadata.creationTimestamp --no-headers -oname)
 
-# Wait for the cluster again to become stable because of remove of pull secret
-wait_till_cluster_stable
+while [ "${mc_before_removing_pullsecret}" == "${mc_after_removing_pullsecret}" ]; do
+	echo "Machine config is still not rendered"
+	mc_after_removing_pullsecret=$(retry ${OC} get mc --sort-by=.metadata.creationTimestamp --no-headers -oname)
+done
```

Comment:
When this loop ends, all operators which can stabilize are either Running or Completed? This is doing far less things than what `wait_till_cluster_stable` was doing.

Reply:
@cfergeau yes but we we need to use the `wait_till_cluster_stable` function then need to ignore the `openshift-marketplace` pods otherwise it will always fail as I mentioned in the commit logs.

Reply:
I understand we cannot use `wait_till_cluster_stable` as is, I'm asking if the checks done by the new code are enough to get the cluster to the state we want?

Reply:
@cfergeau As of now I am depend on the CI for all the changes and if it is green that mean bundle which is created can work with crc and execute the required tests. If we want extra layer of surety then may be I will create another PR to wait again for all the operators except market place but this check also required to make sure MCO took action and new machine config is generated.

Reply:
@cfergeau Added another comment to ignore the openshift-marketplace namespace in the `wait_till_cluster_stable` function. and using along with this.